### PR TITLE
consistently use stdout and stderr

### LINKED ISSUE #444: Certain errors during create/init are outputted only in STOUT not STDERR
### Issue description
When running certain commands that results in an error, such as using the wrong credentials in constellation-conf.yaml or a CSP credentials with not enough privileges.

### To reproduce

1. Run `constellation config generate`
2. Add your credentials in constellation-conf.yaml file (maybe falsify your credentials to trigger an error when exeucting other commands after)
3. Run `constellation create --control-plane-nodes 1 --worker-nodes 1 --name test -y` to try to create a cluster
4. Notice how there is an error in the STDOUT but not in STDERR. (You may need to monitor both STDOUT and STDERR beforehand to properly replicate the bug)
5. This is a sample of the error that could be shown:
```
An error occurred: PUT https://management.azure.com/subscriptions/6b6a1f27-55c1-4b1d-969b-60a3c9eebe64/resourceGroups/azure-customertwo-az-sandbox-84v4y/providers/Microsoft.Insights/components/constellation-insights-nm4rV
--------------------------------------------------------------------------------
RESPONSE 403: 403 Forbidden
ERROR CODE: AuthorizationFailed
--------------------------------------------------------------------------------
{
  "error": {
    "code": "AuthorizationFailed",
    "message": "The client '<ID>' with object id '<ID>' does not have authorization to perform action 'Microsoft.Insights/components/write' over scope '/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP_NAME>/providers/Microsoft.Insights/components/constellation-insights-nm4rV' or the scope is invalid. If access was recently granted, please refresh your credentials."
  }
}
--------------------------------------------------------------------------------

Attempting to roll back.
```

### Environment

- `constellation version`:
- `constellation-conf.yaml`
  - (make sure to remove sensitive information, e.g., `yq e 'del(.provider.*.project)' constellation-conf.yaml`)
- VM type used to run Constellation.

### Expected behavior
As a developer, I expect a failure to output both on STDOUT and STDERR so I can integrate and handle failures more easily

### PR DESCRIPTION
**Pull request by datosh**
Signed-off-by: Fabian Kammel <fk@edgeless.systems>

### Proposed change(s)

- Consistently use
  - stderr for warnings and errors
  - stdout for all other messages

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

**Review by daniel-weisse — APPROVED** [review]
LGTM

File: cli/internal/cmd/create.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func NewCreateCmd() *cobra.Command {
 
 func runCreate(cmd *cobra.Command, args []string) error {
 	fileHandler := file.NewHandler(afero.NewOsFs())
-	spinner := newSpinner(cmd.OutOrStdout())
+	spinner := newSpinner(cmd.ErrOrStderr())
```
[the remark this answers is not in the document]
Comment (Nirusu) [reply cli/internal/cmd/create.go:44]: I think `stderr` is right for this one. For example, curl, tqdm (Python) and go-progressbar (Go) also all print status messages to `stderr`.

[the remark this answers is not in the document]
Comment (datosh) [reply cli/internal/cmd/create.go:44]: I agree! Changed progress bar to write to `stderr` as well!

**Review by 3u13r — APPROVED** [review]
LGTM

**Pull request merged**